src/components/metadata/index.js

```diff
@@ -59,10 +62,7 @@ const MetaData = ({
           <>
             <div className="pipeline-metadata__header-toolbox">
               <div className="pipeline-metadata__header">
-                <NodeIcon
-                  className="pipeline-metadata__icon"
-                  type={metadata.node.type}
-                />
+                <NodeIcon className="pipeline-metadata__icon" icon={icon} />
```

Comment:
looking at the icon assignment here it does look confusing - it would be best to rename the `icon` variable assignment name to something more declarative, like `nodeTypeIcon`